charts/patch/templates/patch.json

```diff
@@ -29,6 +62,12 @@
   },
   {{- end }}
   {{- range $label, $value := .Values.labels }}
+  {{- if $alreadyInjected }}
```

Comment:
Is it possible that the set of labels here during reinvocation is different from the set of a labels from a previous invocation?  If so, we could have some previous labels which are not cleaned up here.  Would it make sense to remove all labels with the `linkerd.io` prefix?

On the other hand, if the label set changing is not a concern, do we need to do the remove at all?  Doing the add should replace the previous value.

Reply:
Yeah, I think it'd make sense to remove all labels prefixed with `linkerd.io` :+1: